shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp

```diff
@@ -353,6 +366,38 @@ void NativeStreamingServerHandler::setUpConfigProtocolCallbacks(std::shared_ptr<
     OnPacketBufferReceivedCallback clientToDeviceStreamingCb = configServerCallbacks.second;
     sessionHandler->setConfigPacketReceivedHandler(receiveConfigPacketCb);
     sessionHandler->setPacketBufferReceivedHandler(clientToDeviceStreamingCb);
+
+    // handle first received config packet with instantiated callback
+    if (receiveConfigPacketCb)
+        receiveConfigPacketCb(std::move(firstPacketBuffer));
+}
+
+void NativeStreamingServerHandler::connectConfigProtocol(std::shared_ptr<ServerSessionHandler> sessionHandler,
+                                                         config_protocol::PacketBuffer&& firstPacketBuffer)
+{
+    {
+        std::scoped_lock lock(sync);
+
+        // Rejects new conenctions in case if count of concurrent config connections exceed the limit
+        // defined with "MaxAllowedConfigConnections" property (where 0 value stands for unlimited count)
+        if (maxAllowedConfigConnections != 0 && configConnectionsCount == maxAllowedConfigConnections)
+        {
+            auto reply = config_protocol::ConfigProtocolServer::generateConnectionRejectedReply(
+                firstPacketBuffer.getId(),
+                OPENDAQ_ERR_CONNECTION_LIMIT_REACHED,
+                "Native configuration connection rejected - connections limit reached on server",
```

Comment:
Usually, the max native configuration connections are either 0 or 1. Is it possible to inject the IP(s) of the already connected clients into the error message?

Reply:
The IP address of the connected client is currently not accessible to the higher level code, but the native library interface could be extended to enable this.

Reply:
This would be great to add, as it eases debugging a lot.